### PR TITLE
[L2B-7768] Add support for ZKSync Era

### PR DESCRIPTION
**Pull request by maciekop-l2b**

### CODE REVIEW

File: packages/uops-dashboard/src/server/clients/code/CodeClient.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export interface CodeClient {
```
Comment (antooni) [inline packages/uops-dashboard/src/server/clients/code/CodeClient.ts:1]: is there a need for this interface? for now we have only type of this client

File: packages/shared/src/uops/protocols/eip-712/const.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const EIP712_TX_TYPE = 113
```
Comment (antooni) [inline packages/shared/src/uops/protocols/eip-712/const.ts:1]: is there a need for seperate file?

File: packages/uops-dashboard/src/server/clients/block/RpcClient.ts [hunk 3]
```diff
@@ -37,7 +37,9 @@ export class RpcClient implements BlockClient {
       transactions: rpcBlock.transactions.map((tx: EVMTransaction) => ({
```
Comment (antooni) [inline packages/uops-dashboard/src/server/clients/block/RpcClient.ts:37]: there is no need for this mapping

File: packages/uops-dashboard/src/server/clients/apiUrls.ts [hunk 4]
```diff
@@ -43,7 +45,7 @@ export function getScanUrl(chainId: ChainId): string {
 }
 
 export function getApiKey(chainId: ChainId, type: ApiKeyType): string {
-  const envName = `${chainId.toUpperCase()}_${type}_API_KEY`
+  const envName = `${chainId.toUpperCase().replace('-', '_')}_${type}_API_KEY`
```
Comment (antooni) [inline packages/uops-dashboard/src/server/clients/apiUrls.ts:48]: why is there a need for this change?

File: packages/shared/src/uops/protocols/eip-712/const.ts (hunk 2 above)
Comment (maciekop-l2b) [reply packages/shared/src/uops/protocols/eip-712/const.ts:1]: wanted to preserve the same structure as for all other protocols

File: packages/uops-dashboard/src/server/clients/apiUrls.ts (hunk 4 above)
Comment (maciekop-l2b) [reply packages/uops-dashboard/src/server/clients/apiUrls.ts:48]: the name of the var is `ZKSYNC_ERA_SCAN_API_KEY` so `zksync-era` get's translated to `ZKSYNC_ERA`

File: packages/uops-dashboard/src/server/clients/block/RpcClient.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (maciekop-l2b) [reply packages/uops-dashboard/src/server/clients/block/RpcClient.ts]: Removed

File: packages/uops-dashboard/src/server/clients/code/CodeClient.ts (hunk 1 above)
Comment (maciekop-l2b) [reply packages/uops-dashboard/src/server/clients/code/CodeClient.ts]: I was planning to introduce 2nd client but there was no need for it in the end. Will remove the interface for now

File: packages/uops-dashboard/src/server/clients/block/RpcClient.ts (hunk 3 above)
Comment (maciekop-l2b) [reply packages/uops-dashboard/src/server/clients/block/RpcClient.ts]: removed

**Review by antooni — APPROVED** [review]

**Pull request merged**